src/daemons/rhsm_d.py

```diff
@@ -170,11 +171,9 @@ def check_status(self):
         """
         log.debug("D-Bus interface com.redhat.SubscriptionManager.EntitlementStatus.check_status called")
         ret = check_status(self.force_signal)
-        if (ret != self.last_status):
-            debug("Validity status changed, fire signal")
-            #we send the code out, but no one uses it at this time
-            self.entitlement_status_changed(ret)
-        self.last_status = ret
+        debug("Validity status changed, fire signal")
+        #we send the code out, but no one uses it at this time
+        self.entitlement_status_changed(ret)
```

Comment:
@alikins  trying this out to see if it fixes @cnsnyder 's issue. if it does I will keep it , else I will revert the latest commit, and go back to comparing with the cached version.